plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +358,27 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*10.
+		if checkOverFlow(totalPriority, config.Weight*10) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * 10
+	}
 	return configs, nil
 }
 
+// checkOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkOverFlow(totalPriority int, weight int) bool {
```

Comment:
The check may be simple enough, my only concern is I have to write unit tests for getPriorityFunctionConfigs, which is kind of orthogonal to issue. IMO, the best thing to do is leave this function as is here now and move it to utils once we find other functions that can use this. I am open to other suggestions too.